src/_guides/libraries/objective-c-interop.md

```diff
@@ -293,6 +293,44 @@ then check the status, and wait for the duration of the audio file:
     }
 ```
 
+### Callbacks and multithreading limitations
+
+While `ffigen` does support converting
+Dart functions to Objective-C blocks,
+there are some limitations to keep in mind
+when using them.
+
+1. Dart isolates are not the same thing as threads.
+   An isolate is not guaranteed to run on a particular thread,
+   and the VM may change which thread an isolate is running on
+   without warning.
+   There is an [open feature request][] to allow isolates to be
+   pinned to specific theads.
+2. Most Apple APIs don't have any guarantees about
+   which thread a callback will be run on.
+3. Many Apple APIs can only be called from the main thread,
```

Comment:
I would say
> `Many Apple APIs are [not thread safe](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Multithreading/ThreadSafetySummary/ThreadSafetySummary.html). In addition, most APIs that involve UI interaction can only be called on the main thread, also known in Flutter as the platform thread.`

Because of the lack of thread pinning, even using a non-main-thread-only object for across multiple calls is not safe if I'm understanding point 1 correctly.

Reply:
I think not thread hostile is good enough. Assuming that the user is interacting with the object from a single isolate, that isolate is only going to be run on one thread at a time. But yeah, I'll reword this to capture that.